### PR TITLE
IGAPP-645: Move layout module to web-ts

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
This only moves the layout module to web-ts. Some todos are already fixed as well as layout components used in #305.

File: web-ts/src/components/GeneralHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import buildConfig from '../constants/buildConfig'
+
+type PropsType = {
+  viewportSmall: boolean
+  t: TFunction
+}
+
+const GeneralHeader = ({ viewportSmall, t }: PropsType) => {
+  // const getPath = new I18nRedirectRouteConfig().getRoutePath
+  // TODO use right path
```
Comment (maxammann) [inline web-ts/src/components/GeneralHeader.tsx:15]: I'm,  a little bit afaid we will miss some of the todos if there are no issue link. We could also use TODO-REFACTORING or FIXME from now on to keep track.

**Review by maxammann — APPROVED** [review]

File: web-ts/src/components/GeneralHeader.tsx (hunk 1 above)
Comment (klinzo) [reply web-ts/src/components/GeneralHeader.tsx:15]: We got it :) We can just check IntelliJ TODOs tab and search afterwards. I have it on my TODO list as well :grin: 
This is also already fixed in #305.

File: web-ts/src/components/HeaderTitle.tsx [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-import type { ThemeType } from 'build-configs/ThemeType'
-import dimensions from '../../theme/constants/dimensions'
+import styled from 'styled-components'
+import dimensions from '../constants/dimensions'
 
-type PropsType = {|
-  children: ?string
-|}
+type PropsType = {
+  children: string | undefined
```
Comment (Taggotty) [inline web-ts/src/components/HeaderTitle.tsx:6]: personally I prefer `children?: string`, just as it looks better. You decide.

File: web-ts/src/components/__tests__/LocationHeader.spec.tsx [hunk 3]
```diff
@@ -59,12 +50,12 @@ describe('LocationHeader', () => {
...
-  const language = 'de'
-  const city = 'augsburg'
-  const location = route => createLocation({ type: route, payload: { city, language } })
+  const languageCode = 'de'
+  const cityCode = 'augsburg'
+  const pathname = '/augsburg/de/willkommen'
   const onStickyTopChanged = (value: number) => {}
 
-  type GetByTextType = (text: string, options?: {| exact: boolean |}) => boolean
+  type GetByTextType = (text: Matcher, options?: SelectorMatcherOptions | undefined) => HTMLElement
```
Comment (Taggotty) [inline web-ts/src/components/__tests__/LocationHeader.spec.tsx:58]: It should be enough to have either `?` or `undefined` 
```suggestion
  type GetByTextType = (text: Matcher, options?: SelectorMatcherOptions) => HTMLElement
```

**Review by Taggotty — APPROVED** [review]

File: web-ts/src/components/Layout.tsx [hunk 4]
```diff
@@ -98,17 +95,17 @@ const Aside = withPlatform(styled.aside`
...
+`
+
+type PropsType = {
+  asideStickyTop: number
+  footer?: ReactNode
+  header?: ReactNode
+  toolbar?: ReactNode
+  modal?: ReactNode
+  children?: ReactNode
   darkMode?: boolean
```
[the remark this answers is not in the document]
Comment (klinzo) [reply web-ts/src/components/Layout.tsx:107]: Removed already in #305

File: web-ts/src/components/LocationLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+import { CategoriesMapModel, CityModel } from 'api-client'
+
+export type FeedbackRatingType = 'up' | 'down'
+
+type PropsType = {
+  cities: Array<CityModel> | null
+  categories: CategoriesMapModel | null
+  viewportSmall: boolean
+  children?: ReactNode
+  toggleDarkMode: () => void
```
[the remark this answers is not in the document]
Comment (klinzo) [reply web-ts/src/components/LocationLayout.tsx:14]: Removed already in #305

File: web-ts/src/components/GeneralHeader.tsx (hunk 1 above)
Comment (klinzo) [reply web-ts/src/components/GeneralHeader.tsx:15]: Created the following issue for other todos regarding checking whether url matches a route: https://issues.tuerantuer.org/browse/IGAPP-668

File: web-ts/src/components/HeaderTitle.tsx (hunk 2 above)
Comment (klinzo) [reply web-ts/src/components/HeaderTitle.tsx:6]: Will fix in #305

File: web-ts/src/components/__tests__/LocationHeader.spec.tsx (hunk 3 above)
Comment (klinzo) [reply web-ts/src/components/__tests__/LocationHeader.spec.tsx:58]: Will fix in #305

**Pull request merged**